client/src/components/TheAnalyticsDemoForm.vue

```diff
@@ -0,0 +1,110 @@
+<template>
+    <form @submit.prevent="submitForm">
+        <v-select v-model="form.date" :items="selectValues.date" item-text="text" item-value="value" label="Date" />
+
+        <v-select
+            v-model="form.cameFrom"
+            :items="selectValues.cameFrom"
+            item-text="text"
+            item-value="value"
+            label="Came From"
+        />
+
+        <v-select v-model="form.user" :items="selectValues.user" item-text="text" item-value="value" label="User" />
+
+        <v-select
+            v-model="form.action"
+            :items="selectValues.action"
+            item-text="text"
+            item-value="value"
+            label="Action"
+        />
+
+        <v-btn type="submit">Update</v-btn>
+    </form>
+</template>
+
+<script>
+import dayjs from 'dayjs';
+
+export default {
+    props: {
+        maxUsers: {
+            type: Number,
+            required: false,
+            default: 100
+        },
+        maxDaysInFuture: {
+            type: Number,
+            required: false,
+            default: 30
+        }
+    },
+
+    data() {
+        return {
+            form: {
+                date: dayjs().toISOString(),
+                cameFrom: 'facebook',
+                user: 'user1',
+                action: 'homepage'
+            },
+            selectValues: {
+                date: (() => {
+                    const dates = [];
```

Comment:
build it above `return` it will be more clear
also why is it called `date` but you return `dates`?

Reply:
@daniel-pupczyk-rmtcfm-com Well. It is `selectValues.date`, means selecValues of the date. That's why i'm returning `dates` and assign it to `date`. But maybe I'm gonna be more clear.